### ISSUE TITLE
Follow points don't respect audio device offset

### ISSUE THREAD

**Issue by WebFreak001** [ppy/osu#2196]
Basically if you change your global audio offset to like 200ms all follow points between circles won't be delayed with the circles.

I managed to find out that I need to add the offset (eg 200ms offset -> `+ 200`) to these two variables (or just add to starttime, but I think those make a bit more sense): https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Osu/Objects/Drawables/Connections/FollowPointRenderer.cs#L84-L85

However it seems like getting the raw settings value is not used by anything, so I wouldn't use that. I managed to find out that Player.cs creates an offset clock wrapping some other clock which presumably does something, but there doesn't seem to be any way to get that instance as it has been recently refactored into a local variable and just passed via a constructor to other fields. I couldn't find any way to pass that Clock to the OsuPlayfield.cs which creates the `FollowPointRenderer` from the Player.cs without major refactoring. There are several classes depending on `OsuPlayfield`, so changing the default constructor would not be an option.

I just added a constant `+ 200` for my game there because it resembles my settings, maybe this will help someone else who has a problem with the follow points.

TLDR: if you use an audio offset, edit these two lines linked above and add your offset to them for nice follow points.

**Comment by WebFreak001** [comment]
closing as I don't want to spend any more time on this and I am only supposed to make issues I personally fix.

**Comment by peppy** [comment]
That’s not true. Someone else will work in a fix for this.

**Comment by WebFreak001** [comment]
ISSUE_TEMPLATE.md:
> [...] Please avoid creating issues or bugs if you do not personally intend to fix them. Some acceptable topics include:
> * [...]
> * Bugs that you have found and are personally willing and able to fix

**Comment by peppy** [comment]
This needs an update, we now openly accept bug reports.

**Comment by WebFreak001** [comment]
oh ok cool, I wasn't really a fan of how emphasized that guideline was. These issue guidelines would prevent me from posting what I was planning on doing because if I then couldn't continue like here I would just have to throw away all the work, which was discouraging in general. I can understand a bit why it was set in the past, but imo it is too binding for the dev. If someone wants to make a small change it might have worked except if they get stuck, but once someone would need to make a big change requiring some refactoring there wouldn't really have been any place to discuss this other than discord, which requires a previous "sign up process" in the #arrival channel (again, feeling like you are then obligated to do work)

I can see that many people would have made issues like "nothing works" if they were here just to play a new osu, but I think most devs would have just posted concrete issues (not missing things) that were applicable at that time. They might give it a try fixing it but if they fail it's not a problem then because someone else might pick it up later. This also makes issues a great source of information, the actual current state, and a place to share some ongoing information about an issue. All while not feeling to bound to personally fixing it.

/rant end

**Comment by smoogipoo** [comment]
Not sure how you're seeing this - it looks fine to me... Both the hitobjects and the followpointrenderer get the same clock so I'm not sure how this would be possible.

**Comment by peppy** [comment]
@WebFreak001 could you please add a video showing this happening so we can try and understand it?

**Comment by WebFreak001** [comment]
hm I mean audio in general is completely offset from music by over 200ms on linux (like you can record it and clearly see it), since the very start of the project, maybe that could be the reason. I will post a video later today.

**Comment by WebFreak001** [comment]
hm now xbuild doesn't work anymore with the project (it can't parse the project files for some reason) and the deprecation message on it got even more intense, but as building with msbuild doesn't work on linux (at least I have tried many different ways, wasting hours on this without documentation) I can't provide any video anymore because I lost the last build I had of this.

Errors with msbuild ([whole log](https://wfr.moe/fmLv7E)):
```
...

"/home/webfreak/dev/osu/osu.sln" (default target) (1) ->
"/home/webfreak/dev/osu/osu.Game/osu.Game.csproj" (default target) (2) ->
"/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj" (default target) (3:2) ->
(CoreCompile target) -> 
  Graphics/Textures/RawTexture.cs(8,42): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(27,44): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(27,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(36,42): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(36,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(23,43): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(23,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(25,43): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(25,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(45,43): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(45,57): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(45,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(63,39): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(63,53): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(63,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(98,44): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(98,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(83,43): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(83,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(90,42): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/Color4Extensions/Color4Extensions.cs(90,23): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/ExtensionMethods.cs(154,54): error CS0246: The type or namespace name 'Size' could not be found (are you missing a using directive or an assembly reference?) [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Extensions/ExtensionMethods.cs(165,58): error CS0246: The type or namespace name 'SecureString' could not be found (are you missing a using directive or an assembly reference?) [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]

...

  Graphics/Vector2Extensions.cs(49,41): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Graphics/Vector2Extensions.cs(49,59): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Graphics/Vector2Extensions.cs(60,45): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Graphics/Vector2Extensions.cs(60,59): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Graphics/Vector2Extensions.cs(72,48): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Graphics/Vector2Extensions.cs(72,66): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Graphics/Visualisation/InfoOverlay.cs(33,48): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]
  Graphics/Visualisation/LogOverlay.cs(213,17): error CS0012: The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj]

    0 Warning(s)
    639 Error(s)
```

I installed msbuild-stable, dotnet-host, dotnet-runtime, dotnet-sdk from arch repos.

With xbuild this happens:

```

>>>> xbuild tool is deprecated and will be removed in future updates, use msbuild instead <<<<

XBuild Engine Version 14.0
Mono, Version 5.8.0.108
Copyright (C) 2005-2013 Various Mono authors

Build started 3/29/2018 5:46:27 PM.
__________________________________________________
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game/osu.Game.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu-resources/osu.Game.Resources/osu.Game.Resources.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.

...

Warnings:

/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game/osu.Game.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu-resources/osu.Game.Resources/osu.Game.Resources.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Rulesets.Osu/osu.Game.Rulesets.Osu.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Rulesets.Catch/osu.Game.Rulesets.Catch.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Rulesets.Taiko/osu.Game.Rulesets.Taiko.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Rulesets.Mania/osu.Game.Rulesets.Mania.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Desktop.Deploy/osu.Desktop.Deploy.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Tests/osu.Game.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Desktop/osu.Desktop.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Rulesets.Catch.Tests/osu.Game.Rulesets.Catch.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Rulesets.Mania.Tests/osu.Game.Rulesets.Mania.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Rulesets.Taiko.Tests/osu.Game.Rulesets.Taiko.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : /home/webfreak/dev/osu/osu.Game.Rulesets.Osu.Tests/osu.Game.Rulesets.Osu.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
/home/webfreak/dev/osu/osu.sln:  warning : Don't know how to handle GlobalSection ExtensibilityGlobals, Ignoring.

Errors:

/home/webfreak/dev/osu/osu.sln (default targets) ->
(Build target) ->

	/home/webfreak/dev/osu/osu.Game/osu.Game.csproj: error : /home/webfreak/dev/osu/osu.Game/osu.Game.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj: error : /home/webfreak/dev/osu/osu-framework/osu.Framework/osu.Framework.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu-resources/osu.Game.Resources/osu.Game.Resources.csproj: error : /home/webfreak/dev/osu/osu-resources/osu.Game.Resources/osu.Game.Resources.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Rulesets.Osu/osu.Game.Rulesets.Osu.csproj: error : /home/webfreak/dev/osu/osu.Game.Rulesets.Osu/osu.Game.Rulesets.Osu.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Rulesets.Catch/osu.Game.Rulesets.Catch.csproj: error : /home/webfreak/dev/osu/osu.Game.Rulesets.Catch/osu.Game.Rulesets.Catch.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Rulesets.Taiko/osu.Game.Rulesets.Taiko.csproj: error : /home/webfreak/dev/osu/osu.Game.Rulesets.Taiko/osu.Game.Rulesets.Taiko.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Rulesets.Mania/osu.Game.Rulesets.Mania.csproj: error : /home/webfreak/dev/osu/osu.Game.Rulesets.Mania/osu.Game.Rulesets.Mania.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Tests/osu.Game.Tests.csproj: error : /home/webfreak/dev/osu/osu.Game.Tests/osu.Game.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Desktop/osu.Desktop.csproj: error : /home/webfreak/dev/osu/osu.Desktop/osu.Desktop.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Rulesets.Catch.Tests/osu.Game.Rulesets.Catch.Tests.csproj: error : /home/webfreak/dev/osu/osu.Game.Rulesets.Catch.Tests/osu.Game.Rulesets.Catch.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Rulesets.Mania.Tests/osu.Game.Rulesets.Mania.Tests.csproj: error : /home/webfreak/dev/osu/osu.Game.Rulesets.Mania.Tests/osu.Game.Rulesets.Mania.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Rulesets.Taiko.Tests/osu.Game.Rulesets.Taiko.Tests.csproj: error : /home/webfreak/dev/osu/osu.Game.Rulesets.Taiko.Tests/osu.Game.Rulesets.Taiko.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  
	/home/webfreak/dev/osu/osu.Game.Rulesets.Osu.Tests/osu.Game.Rulesets.Osu.Tests.csproj: error : /home/webfreak/dev/osu/osu.Game.Rulesets.Osu.Tests/osu.Game.Rulesets.Osu.Tests.csproj: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.  

	 15 Warning(s)
	 13 Error(s)
```

build commands:
```sh
# Mono JIT compiler version 5.10.0 (makepkg/dac60775616 Thu Mar 22 02:05:14 CET 2018)
# NuGet Version: 4.4.1.4656
# Microsoft (R) Build Engine version 15.4.0.0 ( Wed Nov 29 14:56:11 UTC 2017) for Mono
# cloned recursively.

nuget restore

msbuild /p:Configuration=Release
# or
xbuild /p:Configuration=Release
```

It is an improvement that there is finally a COMPILING.md file and I followed the steps on there, have all the dependencies installed but these issues have happened since the beginning and are still persisting, I see people workaround it but it is never brought up and now it is actually completely blocking me from compiling the game as the xbuild workaround doesn't seem to work anymore.

**Comment by WebFreak001** [comment]
ok update: I managed to finally make it work by doing this, so if you are on archlinux this might help you:
* uninstall dotnet, nuget4, nuget5, dotnet-host, dotnet-runtime, dotnet-sdk, they all interfere with the build process
* install mono and msbuild-stable, delete all other msbuild packages you might have
* do a fresh recursive clone of osu, `rm -rf ~/.nuget`, `nuget restore`
* `export TERM=xterm` before build in case you get weird console magic number mismatch errors
* `msbuild /p:Configuration=Release` or `msbuild /p:Configuration=Debug`
* `mono osu.Desktop/bin/Release/net461/osu!.exe`

**Comment by WebFreak001** [comment]
gonna close as the actual problem described in the issue doesn't exist.

**Comment by WebFreak001** [comment]
actually I just found out that this issue persists, but it isn't bound to local offset, it's bound to audio device offset (or well, however osu syncs the beatmap with the audio device).

I managed to find out my old configuration (PulseAudio as audio device) which had the 200ms offset. See this video with 0ms audio offset in the settings, it has the delay in the follow points, changing the audio offset to match the music does actually not make a difference: [https://wfr.moe/fmcYeM.mp4](https://wfr.moe/fmcYeM.mp4)

**Comment by peppy** [comment]
You made two videos for different beatmaps which makes it near impossible to compare.

**Comment by WebFreak001** [comment]
well the audio offset doesn't change anything, I tried -500ms and +500ms and both looked the same where both follow points are always a bit early. I updated the title of this issue to reflect that, it didn't happen with the ALSA sound device (or at least it was not noticable) which had only like 70ms offset, but it is noticable with the pulseaudio device which has around 220ms offset.

Now I did some more tests and found out that this is actually really hard to reproduce, I made a small test map but it looks normal, in the new editor the follow lines are if anything just offset by around 80ms (they were there about 3/4 ticks on a 180bpm map earlier than in the normal osu editor, but after restarts and playing and pausing a lot this seems to vary a lot), but it does actually feel like it's a bit late _at some points in some maps_

Now here is the same recording as above just with an offset of 220ms: [https://wfr.moe/fmsv7x.mp4](https://wfr.moe/fmsv7x.mp4) and for reference in normal osu [https://wfr.moe/fmB5Zv.mp4](https://wfr.moe/fmB5Zv.mp4)

<details>

![https://wfr.moe/fms3RE.png](https://wfr.moe/fms3RE.png)
Like the dots are basically there in near full opacity, while on the osu play they still need a few recording frames to reach that opacity.

![https://wfr.moe/fmszWS.png](https://wfr.moe/fmszWS.png)
here one is already visible while on the osu play it's only visible around 3 recording frames later

</details>

So actually after looking at it frame by frame and doing some checks in the editor, the positions match the osu play, but the opacity is like 50-100ms delayed

**Comment by WebFreak001** [comment]
hm I guess this is rather a new design decision rather than there being an offset, it's not a big gameplay issue, but for me it felt a lot better if I added a bit of offset to the follow points using that code up there.
